src/main/java/dev/j3fftw/litexpansion/weapons/NanoBlade.java

```diff
@@ -29,12 +31,17 @@ public float getMaxItemCharge(ItemStack item) {
     @Override
     public ItemUseHandler getItemHandler() {
         return event -> {
+            ItemMeta nanoBladeMeta = event.getItem().getItemMeta();
             final Enchantment enchantment = Enchantment.getByKey(Constants.NANO_BLADE_ACTIVE_ENCHANT);
-            final int removedLevel = event.getItem().removeEnchantment(enchantment);
 
-            if (removedLevel == 0) {
-                event.getItem().addEnchantment(enchantment, 1);
+            if (!nanoBladeMeta.hasEnchant(enchantment)) {
+                nanoBladeMeta.addEnchant(enchantment, 1, false);
+                nanoBladeMeta.setDisplayName(ChatColor.DARK_GREEN + "Nano Blade" + ChatColor.GREEN + " (On)");
+            } else {
+                nanoBladeMeta.removeEnchant(enchantment);
```

Comment:
Not sure why you moved this. In ItemMeta IIRC removeEnchant returns a boolean so you can just do it in the if statement